src/libexpr/nixexpr.cc

```diff
@@ -105,7 +105,7 @@ void ExprAttrs::showBindings(const SymbolTable & symbols, std::ostream & str) co
     }
     for (const auto & [from, syms] : inheritsFrom) {
```

Comment:
This could be considered the root cause.
Indexing by a pointer is not necessarily wrong, but enumerating items ordered by addresses is non-deterministic.